### PR TITLE
Add common and tag dimensions across compliance queries.  Closes #90

### LINKED ISSUE #90: Add common and tag dimensions across compliance queries
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

Common & tag dimensions will allow the end user with options to render compliance output based on tags key, connection name, project-id & region.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**

### CODE REVIEW

File: policy_bundle/kubernetes.sp [hunk 1]
```diff
@@ -76,7 +76,7 @@ control "enable_auto_repair" {
 
 control "enable_auto_upgrade" {
   title = "Ensure Automatic node upgrades is enabled on Kubernetes Engine Clusters nodes"
-  query = query.kubernetes_cluster_auto_repair_enabled
+  query = query.kubernetes_cluster_auto_upgrade_enabled
```
Comment (rajlearner17) [inline policy_bundle/kubernetes.sp:79]: @Priyanka-Chatterjee-2000 did we change this query?

**Review by rajlearner17 — APPROVED** [review]
LGTM as per latest change and testing

**Review by misraved — COMMENTED** [review]
@Priyanka-Chatterjee-2000 I have left a few review comments. Please let me know if you have questions. Thanks!!

File: README.md [hunk 2]
```diff
@@ -86,6 +86,31 @@ This mod uses the credentials configured in the [Steampipe GCP plugin](https://h
 
 No extra configuration is required.
 
+### Common and Tag Dimensions
+
+The benchmark queries use common properties (like `connection_name`, `name`, `organization_id`,  `location` and ` project`) and tags that are defined in the form of a default list of strings in the `mod.sp` file. These properties can be overwritten in several ways:
```
Comment (misraved) [inline README.md:91]:
```suggestion
The benchmark queries use common properties (like `connection_name`, `organization_id`,  `location` and ` project`) and tags that are defined in the form of a default list of strings in the `mod.sp` file. These properties can be overwritten in several ways:
```

File: mod.sp [hunk 3]
```diff
@@ -7,6 +7,66 @@ locals {
...
+  EOQ
+
+  common_dimensions_qualifier_organization_sql = <<-EOQ
+  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{endif~}
+  %{~if contains(var.common_dimensions, "project")}, __QUALIFIER__organization_id%{endif~}
+  EOQ
+
+  common_dimensions_qualifier_project_sql = <<-EOQ
+  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{endif~}
+  %{~if contains(var.common_dimensions, "project")}, __QUALIFIER__name%{endif~}
```
Comment (misraved) [inline mod.sp:50]:
```suggestion
  %{~if contains(var.common_dimensions, "project")}, __QUALIFIER__name as project%{endif~}
```

File: mod.sp [hunk 4]
```diff
@@ -7,6 +7,66 @@ locals {
...
+  default = []
+}
+
+locals {
+  # Local internal variable to build the SQL select clause for common
+  # dimensions using a table name qualifier if required. Do not edit directly.
+  common_dimensions_qualifier_sql = <<-EOQ
+  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name'%{endif~}
+  %{~if contains(var.common_dimensions, "location")}, __QUALIFIER__location%{endif~}
+  %{~if contains(var.common_dimensions, "project")}, __QUALIFIER__project%{endif~}
```
Comment (misraved) [inline mod.sp:35]:
```suggestion
  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{endif~}
  %{~if contains(var.common_dimensions, "location")}, __QUALIFIER__location as location%{endif~}
  %{~if contains(var.common_dimensions, "project")}, __QUALIFIER__project as project%{endif~}
```

**Review by misraved — COMMENTED** [review]
@Priyanka-Chatterjee-2000 added a few more review comments. Please take a look!!

File: forseti_security_v226/forseti_security.sp [hunk 5]
```diff
@@ -15,7 +15,7 @@ benchmark "forseti_security_v226" {
     control.iam_restrict_service_account_key_age_one_hundred_days,
     control.only_my_domain,
     control.require_bq_table_iam,
-    control.restrict_firewall_rule_world_open,
+    control.restrict_firewall_rule_ssh_world_open,
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply forseti_security_v226/forseti_security.sp:18]: Two controls were using same queries and checking the same conditions so removed one.

File: mod.sp [hunk 6]
```diff
@@ -7,6 +7,66 @@ locals {
   }
 }
 
+variable "common_dimensions" {
+  type        = list(string)
+  description = "A list of common dimensions to add to each control."
+  # Define which common dimensions should be added to each control.
+  # - connection_name (_ctx ->> 'connection_name')
+  # - location
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply mod.sp:15]: Removed organization dimension as discussed, it was not used across all, only one query was using it.

**Review by misraved — APPROVED** [review]

**Pull request merged**